src/web/app/components/response-time-series-chart/response-time-series-chart.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, Input, OnInit } from '@angular/core';
+import * as d3 from 'd3';
+import { ResponseTimeSeriesChartModel } from "./response-time-series-chart.model";
+import { FeedbackResponseStatsService } from "../../../services/feedback-response-stats.service";
+import { BehaviorSubject, interval } from "rxjs";
+import { switchMap, tap } from "rxjs/operators";
+import { FeedbackResponseStats } from "../../../types/api-output";
+import { ErrorMessageOutput } from "../../error-message-output";
+
+/**
+ * Response-tracking time series chart for admin user.
+ */
+@Component({
+  selector: 'tm-response-time-series-chart',
+  templateUrl: './response-time-series-chart.component.html',
+  styleUrls: ['./response-time-series-chart.component.scss']
+})
+export class ResponseTimeSeriesChartComponent implements OnInit {
+
+  @Input()
+  model: ResponseTimeSeriesChartModel = {
+    durationMinutes: 60,
+    intervalMinutes: 1
+  }
+
+  interval: BehaviorSubject<number> = new BehaviorSubject<number>(1);
+
+  constructor(private feedbackResponseStatsService: FeedbackResponseStatsService) { }
+
+  ngOnInit(): void {
+    this.interval.pipe(
```

Comment:
I think all code related to rxjs should be put into ```response-time-series-chart.service``` to have clearer abstraction

Reply:
i think this should be fine? I saw it in other components too. The service class is mainly for housing code related to API calls, but I need the rxjs library to do some asynchronous stuff in the component

Reply:
yes I understand. But you can return rxjs Subject from service to component right? So instead of <<block of code>>.subsribe(), we can just have rxjsSubject.subcribe()

Reply:
^ I think both ways are valid. It also depends on how big this piece of code is, so I'd say very subjective. @jianhandev if you are free, you can try moving it to service and see if the code becomes cleaner. I am personally fine with leaving this here as long as it doesn't get unnecessarily messy for viewing.

Reply:
alright, will do more refactoring next week 👍